examples/federated-learning/tff/nvflare/README.md

```diff
@@ -0,0 +1,297 @@
+# NVFlare with TensorFlow
+
+This example uses NVFlare to train an image classifier using federated averaging and TensorFlow as the deep learning framework.
+
+[NVIDIA FLARE](https://nvflare.readthedocs.io/en/main/index.html) (**NV**IDIA **F**ederated **L**earning **A**pplication **R**untime **E**nvironment)
+is a domain-agnostic, open-source, extensible SDK that allows researchers and data scientists to adapt existing ML/DL workflows to a federated paradigm.
+It enables platform developers to build a secure, privacy-preserving offering for a distributed multi-party collaboration.
+
+[Here](https://nvflare.readthedocs.io/en/main/flare_overview.html#high-level-system-architecture) is a high level system architecture of NVFlare.
+
+This example builds on top of the infrastructure that the
+[blueprint provides](../../../../README.md), and follows the best practices the
+blueprint establishes.
+
+## Architecture
+
+The following diagram shows one server and two clients that are connected to the server through a secure gRPC link:
+
+![alt_text](../../../../assets/nvflare.svg "Infrastructure overview")
+
+As shown in the preceding diagram, the blueprint helps you to create and configure the following infrastructure components:
+
+- A persistent volume to store the NVFlare workspace
+- Two pods that are the clients that will be connected to the server in the nvidia-client1 and nvidia-client2 namespaces respectively
+- One pod that is the the server that will aggregate all the results from the computation in the nvflare-infra namespace
+
+## Deploy the blueprint
+
+This example builds on top of the infrastructure that the
+[blueprint provides](../../../../README.md), and follows the best practices the
+blueprint establishes.
+
+To deploy the NVFlare demo described in this document, you need to deploy the [Federated learning blueprint](../../../../README.md#deploy-the-blueprint) first. Then, you can deploy the NVFlare demo described in this document.
+
+### Build and push the example container image
+
+You have to create a custom image with TensorFlow and NVFlare installed. From the main `terraform` folder:
+
+1. Open [Cloud Shell](https://cloud.google.com/shell)
+
+1. Build the example container image locally on your host. It takes about 8GBs of space and can take up to 20mn to build. For a production deployment, consider using Cloud Build:
```

Comment:
Move `It takes about 8GBs of space and can take up to 20mn to build. For a production deployment, consider using Cloud Build` after the command otherwise it confuses readers about what we're doing next.

Reply:
Done